openshift-serverless/tasks/main.yml

```
@@ -0,0 +1,204 @@
+    api_version: operator.knative.dev/v1beta1
+    kind: KnativeEventing
+    definition:
+      metadata:
+        labels:
+          opdev: opcap
+    state: present
+- name: Test knative eventing conditions type Ready
+  kubernetes.core.k8s_info:
+    namespace: knative-eventing
+    name: knative-eventing
+    api_version: operator.knative.dev/v1beta1
+    kind: knativeeventing
+    wait: yes
+    wait_condition:
+      status: "True"
+      type: "Ready"
+    wait_timeout: 90
+  register: knative_eventing
+  failed_when: (knative_eventing.resources[0].status.conditions | selectattr('type', 'search', 'Ready') | list | last).status != "True"
+# 2. Testing knative serving
+- name: Create knative serving
+  kubernetes.core.k8s:
+    name: knative-serving
+    namespace: knative-serving
+    api_version: operator.knative.dev/v1beta1
+    kind: KnativeServing
+    definition:
+      metadata:
+        labels:
+          opdev: opcap
+    state: present
+- name: Test knative serving conditions type Ready
+  kubernetes.core.k8s_info:
+    namespace: knative-serving
+    name: knative-serving
+    api_version: operator.knative.dev/v1beta1
+    kind: knativeserving
+    wait: yes
+    wait_condition:
+      status: "True"
+      type: "Ready"
+    wait_timeout: 90
+  register: knative_serving
+  failed_when: (knative_serving.resources[0].status.conditions | selectattr('type', 'search', 'Ready') | list | last).status != "True"  
+# 3. Testing knative service
+- name: Create knative hello service
+  kubernetes.core.k8s:
+    name: hello
+    namespace: knative-serving
+    api_version: serving.knative.dev/v1
+    kind: Service
+    definition:
+      metadata:
+        labels:
+          opdev: opcap
+      spec:
+        template:
+          spec:
+            containers:
+              - image: docker.io/openshift/hello-openshift
+                env:
+                  - name: RESPONSE
+                    value: "Hello Serverless!"            
+    state: present
+- name: Get hello service Conditions "Ready"
+  kubernetes.core.k8s_info:
+    namespace: knative-serving
+    name: hello
+    api_version: serving.knative.dev/v1
+    kind: Service
+    wait: yes
+    wait_condition:
+      status: "True"
+      type: "Ready"
+    wait_timeout: 90
+- name: Get hello service Conditions "RoutesReady"
+  kubernetes.core.k8s_info:
+    namespace: knative-serving
+    name: hello
+    api_version: serving.knative.dev/v1
+    kind: Service
+    wait: yes
+    wait_condition:
+      status: "True"
+      type: "RoutesReady"
+    wait_timeout: 90
+  register: service
+  failed_when: 
+    - (service.resources[0].status.conditions | selectattr('type', 'search', 'Ready') | list | last).status != "True"
+    - (service.resources[0].status.conditions | selectattr('type', 'search', 'RoutesReady') | list | last).status != "True"
+- name: Check hello service - must return hello serverless
+  ansible.builtin.uri:
+    url: "{{service.resources[0].status.url}}"
```

Jinja spacing should be more like this for readability:

```suggestion
    url: "{{ service.resources[0].status.url }}"
```